frontends/ntt-pipeline/gen-ntt-pipeline.py

```diff
@@ -95,36 +126,36 @@ def fresh_comp_index(op):
 
         return saved_count
 
-    def pp_stage_groups(stage, row, structure_tuple):
-        lhs, op, rhs, phi_index = structure_tuple
-        comp = 'add' if op == '+' else 'sub'
-
-        mult_group_name = 's{}_r{}_mul'.format(stage, row)
+    def pp_mul_group(stage, mul_tuple):
+        mul_index, k, phi_index = mul_tuple
+        mult_group_name = 's{}_mul{}'.format(stage, mul_index)
         mult_wires = [
-            'mult_pipe{}.left = phi{}.out;'.format(row, phi_index),
```

Comment:
Consider using [f-strings](https://docs.python.org/3/tutorial/inputoutput.html):

```
f"mult_pipe{row}.left = phi{phi_index}.out"
```

Reply:
Noted. I'm not going to go through the hassle of changing these now since these will be replaced eventually, but I'll use f-strings in the Python library. 

Not even sure why I transitioned to `str.format` lately; probably from working in the `bril` codebase.